### ISSUE TITLE
Material Shell not working on Gnome 45

### ISSUE THREAD

**Issue by piotron** [material-shell/material-shell#981]
**Describe the bug**
With current release of Gnome 45 Material Shell is not starting with below error

```javascript
SyntaxError: import declarations may only appear at top level of a module

Stack trace:
@file:///usr/share/gnome-shell/extensions/material-shell@papyelgringo/prefs.js:283:12
_init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
```
This is caused [by Gnome](https://blogs.gnome.org/shell-dev/2023/09/02/extensions-in-gnome-45/) moving from GJS to ESM imports.

**Expected behavior**
Material Shell loading.

**Screenshots**
![Screenshot from 2023-10-12 12-32-18](https://github.com/material-shell/material-shell/assets/5966660/ba312621-1fa7-4721-8e3f-de62719e7af9)

**To Reproduce**
- Occurs on:
  - [x] X.Org
  - [x] Wayland
- GNOME Shell version: main
- Distro: Arch Linux/GNOME OS
- Steps to reproduce, if applicable:
1. Upgrade to Gnome 45
2. Material Shell is not loading
**Current commit**
`a322939`

**Log data**
Output from `journalctl /usr/bin/gnome-shell`:
```
Not Applicable
```

**Comment by piotron** [comment]
From what I found [this](https://gjs.guide/extensions/upgrading/gnome-shell-45.html#esm) may also be helpful in fixing the issue.

**Comment by PapyElGringo** [comment]
Yeah I'm aware of the incompatibility, but I said here https://github.com/material-shell/material-shell/discussions/979 I have to make spend time on the future of the project.

But if someone want to look into it :+1: note that we are using Typescript which mean that we cannot just follow the guide you shared but @HalfVoxel could probably guide better on this

**Comment by seantcanavan** [comment]
Upgrading to the latest Manjaro as of today produces an error in the 'Extensions' list which shows Material Shell to be incompatible with the latest version of GNOME. I've rolled-back for now. Fingers crossed!

**Comment by PapyElGringo** [comment]
Wow thanks you very much!
